nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -90,14 +89,14 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
 
         self._validate_trainer()
 
-        if self.trainer.precision == 'bf16':
+        if str(self.trainer.precision)[0:4] == 'bf16':
             self.autocast_dtype = torch.bfloat16
-        elif int(self.trainer.precision) == 32:
+        elif str(self.trainer.precision)[0:2] == '32':
             self.autocast_dtype = torch.float
-        elif int(self.trainer.precision) == 16:
+        elif str(self.trainer.precision)[0:2] == '16':
             self.autocast_dtype = torch.half
         else:
-            raise ValueError('precision must be in [32, 16, "bf16"]')
+            raise ValueError('precision must be in [32, "32-true", 16, "16-mixed", "bf16", "bf16-mixed"]')
```

Comment:
Let's update the error message so it suggest to use the values that won't be deprecated soon:
`["32-true", "16-mixed", "bf16-mixed"]

Reply:
Fixed.